pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1115,6 +1127,23 @@ type PrometheusSpec struct {
 	EnableAdminAPI bool `json:"enableAdminAPI,omitempty"`
 }
 
+type WhenScaledRetentionType string
+
+var (
+	WhenScaledRetentionTypeRetain WhenScaledRetentionType = "Retain"
+	WhenScaledRetentionTypeDelete WhenScaledRetentionType = "Delete"
+)
+
+type ShardRetentionPolicy struct {
+	// WhenScaled defines the retention policy when the Prometheus shards are scaled down.
+	// When set to `Retain`, the operator will keep the pods from the scaled-down shard(s), so the data can still be queried.
+	//
+	// Default: "Delete"
+	// +kubebuilder:validation:Enum=Retain;Delete
+	// +required
```

Comment:
I would define the field as optional for now.
```suggestion
	// +optional
```